main/engines/items.py

```diff
@@ -1,19 +1,80 @@
-from sqlalchemy import func, select
+from collections.abc import Sequence
+
+from sqlalchemy import delete, func, select
 
 from main import db
 from main.models.item import ItemModel
 
 
-async def count_items() -> int:
-    statement = select(func.count()).select_from(ItemModel)
+async def count_items(category_id: int) -> int:
+    statement = (
+        select(func.count())
+        .select_from(ItemModel)
+        .where(ItemModel.category_id == category_id)
+    )
     result = await db.session.execute(statement)
-    return result.scalar()
+    return result.scalar() or 0
+
+
+async def get_items(category_id: int, offset: int, limit: int) -> Sequence[ItemModel]:
+    statement = (
+        select(ItemModel)
+        .where(ItemModel.category_id == category_id)
+        .offset(offset)
+        .limit(limit)
+    )
```

Comment:
We should provide an explicit sorting order here

Reply:
Fixed! Using descending `created` timestamp.